### ISSUE TITLE
git vcs a dependency?

### ISSUE THREAD

**Issue by sillysloft** [AtomLinter/linter-eslint#1065]
## Issue Type
"Question"

Yesterday i've tried to enable atom eslint in a project. It wasn't a real project, but more some testing/fun stuff, so i wasn't using git yet. It was a little weird, only after removing the project folder, an open file would be checked with eslint, otherwise eslint didn't seem to be run at all.
But once i ran `git init .` in the project folder (and maybe a restart of atom) it started working and showing all the errors and warnings in my code.
Now i wonder if git is really a dependency for atom eslint, since it isn't communicated in the readme or anywhere else, or is this probably due to side effects from some other plugin?

**Comment by Arcanemagus** [comment]
> It was a little weird, only after removing the project folder, an open file would be checked with eslint, otherwise eslint didn't seem to be run at all.

I'm not even sure what you mean here 😕.

> (and maybe a restart of atom)

This is far more likely to have been relevant, the only tangential relation to `git` of `linter-eslint` is that `linter` ignores things that are ignored by your VCS... but that isn't `linter-eslint` itself. There is nothing here that depends on a VCS for the project.

**Comment by sillysloft** [comment]
Ok, i've just tried it with a new atom profile (i moved ~/.atom out of the way), installed linter-eslint and it's dependencies, created a new project with a simple eslint configuration (only "extends": "airbnb-base"), created a test.js file and saved it with some code and ran `npm init -y && npm i -D eslint eslint-config-airbnb-base eslint-plugin-import` in the project folder.

Bare minimum to make it work.

Now if i "Add Project Folder" and open the test.js file, it won't lint, then with the file still open, i remove the project folder from the tree-view on the left, suddenly errors in the code are highlighted.
Next, i shutdown atom, run `git init . && echo "node_modules" > .gitignore && git add . && git commit -m "test"`, start atom, "Add Project Folder" open test.js file => **linter-eslint now works just fine**.

Basically:
- `rm -r .git` and restart atom -> eslint stops working in this project
- `git init . && echo "node_modules" > .gitignore && git add . && git commit -m "test"` and restart atom -> eslint starts working as expected

It's definitely not connected to any other atom plugins and very easy to reproduce, i'm running Linux.

I think in #1059 @codequistador describes the same thing happening, when he says "switch back to Project A", actually he removes the project folder and suddenly the file is linted...
I bet he can make it work with git!

**Comment by Arcanemagus** [comment]
How are you opening Atom that you need to "Add Project Folder"?

**Comment by Arcanemagus** [comment]
I'm unable to reproduce by:

1. Open an existing project in Atom, verify `linter-eslint` works as expected.
2. Close Atom
3. Rename `.git` to `.foo`
4. Open Atom from the terminal in the project folder with `atom .`.
    * `linter-eslint` works normally
5. Open a non-project folder (`C:\temp`)
6. Create a new window
7. "Add Project Folder" in the new window for the project from step 1.
8. Verify that `linter-eslint` is working properly.

**Comment by sillysloft** [comment]
Well... dunno, this is what it does here (Linux)...
`mv .git .foo; atom .` all bad
`mv .foo .git; atom .` all good

Do you use windows? (`C:\temp`)

**Comment by sillysloft** [comment]
If it's true at least for Linux, the issue can simply be solved with a comment in the readme, that would have saved me at least half an hour yesterday.

**Comment by Arcanemagus** [comment]
> `mv .git .foo; atom .` all bad

You're going to have to be a _lot_ more specific than that in what you are doing. For an example see my post.

> Do you use windows? (`C:\temp`)

Yes, but that _shouldn't_ have any bearing here.

**Comment by sillysloft** [comment]
Well, i close atom with the close button in window decorations. Then run the command in a bash shell in the project directory. It opens Atom. Then i use the close button again and the other shell command to move .git back in place.
There is nothing more involved :/

**Comment by skylize** [comment]
@sillysloft  What are the outputs of `Linter Eslint: Debug` when it's "all bad" or "all good"?

**Comment by Arcanemagus** [comment]
I'm unable to reproduce this on:

```
OS: Windows 10 x64
Atom    : 1.24.0-beta1
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0
```

As well as unable to reproduce this on:
```
OS: Ubuntu 17.10
Atom    : 1.24.0-beta1
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0
```

Both of them work identically whether the project is under a VCS or not.

**Comment by sillysloft** [comment]
I've installed a new system, Debian 9.x again, there everything is working fine, with and without git.
False alarm then.
Must be something weird with my os then.
Sorry for the fuss.

**Comment by Arcanemagus** [comment]
No worries, marking this as closed but if you ever do figure out what was (is?) going on I'm curious 😉.

**Comment by sillysloft** [comment]
I've tried it again just now, renaming .git to .foo breaks it reliably on the system i currently use.
Finding out what it is would probably involve a few breakpoints in linter and/or linter-eslint i guess.
Well, maybe...

**Comment by Arcanemagus** [comment]
When it's broken can you run `Linter: Debug` _and_ `Linter Eslint: Debug` from the Command Palette?
